### ISSUE TITLE
NULL check for constructor argument of Ical.Net.DataTypes.Organizer

### ISSUE THREAD

**Issue by janousek** [ical-org/ical.net#208]
Constructor of the class Ical.Net.DataTypes.Organizer should check if argument "value" is NULL.
Current behavior: exception is thrown
Expected behaviour: no exception.

Current code:
```
public Organizer(string value) : this()
{
	var serializer = new OrganizerSerializer();
	CopyFrom(serializer.Deserialize(new StringReader(value)) as ICopyable);
}
```

Proposed change:
```
public Organizer(string value) : this()
{
	if(value != null){ 
		var serializer = new OrganizerSerializer();
		CopyFrom(serializer.Deserialize(new StringReader(value)) as ICopyable);
	}
}
```

**Comment by rianjs** [comment]
#187 is related to this, but I like this ticket better, so I'm going to close it and use this one instead.

**Comment by rianjs** [comment]
Setting Attachments = null shouldn't throw an error message. Modifying a calendar event such that it no longer has a collection of attachments is a reasonable thing to want to do.

.Clear() is followed by .AddRange(), but there should be a null check in the middle to bail out if the incoming IEnumerable is null.

https://github.com/rianjs/ical.net/blob/f1002f1b17499f8ed71bb1c63d9718da854d3b98/v2/ical.NET/General/CalendarProperty.cs#L130

**Comment by rianjs** [comment]
All `CalendarComponent` types (`Event`, `Alarm`, `Journal`, etc)  should be evaluated for not exploding when properties are set to null:

* In the constructor
* During object initialization (for public properties)
* When setting a public property to null

Notes: as a general rule, the `CalendarComponent` subtypes map pretty well to [RFC-5545's "Component" section](https://tools.ietf.org/html/rfc5545#section-3.6).

I would prefer to always initialize to an empty collection rather than null. So if `null` is passed to a setter, I'd rather set the underlying field to `Enumerable.Empty<T>()`. Same for constructor initialization.

**Comment by Poojith** [comment]
Hi @rianjs , this is my first time with OSS. I would love to work on the issue.

**Comment by rianjs** [comment]
Go right ahead. Please include a unit test to assert that nulls don't throw exceptions in the places you change.

**Comment by beriniwlew** [comment]
@rianjs , I've got a couple of these fixed, with unit tests added. 

I am also fairly new to OSS, and git/github in general, but I'd like to help.

**Comment by rianjs** [comment]
Fixed in nuget version 4.0.0: https://www.nuget.org/packages/Ical.Net/4.0.0

**Comment by rianjs** [comment]
@beriniwlew - I noticed these fixes as well, thanks!
